source/javascripts/components/common/RepoYmlStorageActions.tsx

```diff
@@ -1,18 +1,59 @@
-import { useState, useMemo } from "react";
+import { useState, useEffect, useRef } from "react";
 import { Box, Icon, Text, Notification, Link } from "@bitrise/bitkit";
 import CopyToClipboard from "react-copy-to-clipboard";
 import { AppConfig } from "../../models/AppConfig";
 import appConfigAsYml from "../../utils/appConfigAsYml";
+import useMonolithApiCallback from "../../hooks/api/useMonolithApiCallback";
 
 type RepoYmlStorageActionsProps = {
 	appConfig: AppConfig | string;
 };
 
+const useFormattedYml = (appConfig: AppConfig): { yml: string; failed: boolean; loading: boolean } => {
+	const [yml, setYml] = useState(typeof appConfig === "string" ? appConfig : "");
+	const formatAppConfigRef = useRef<() => void>();
+	const { loading, failed, result, call } = useMonolithApiCallback<string>(
+		"/api/cli/format",
+		{
+			method: "POST",
+			body: JSON.stringify(appConfig)
+		},
+		(value: string) => value
+	);
+
+	// NOTE: call function isn't referentially stable
+	useEffect(() => {
+		formatAppConfigRef.current = call;
+	});
+
+	useEffect(() => {
+		if (typeof appConfig === "object") {
+			formatAppConfigRef.current?.();
+		}
+	}, [appConfig]);
+
+	useEffect(() => {
+		if (failed && appConfig) {
+			setYml(appConfigAsYml(appConfig));
+		}
+
+		if (result && !failed) {
+			setYml(result);
+		}
+	}, [result, failed, appConfig]);
+
+	return {
+		yml,
+		failed: !!failed,
+		loading
```

Comment:
`failed` and `loading` are not used anywhere

Reply:
Removed them, now just returning the yml